### PR TITLE
feat(build-system): Add pdm as build-system/dependency manager

### PR DESCRIPTION
**Pull request by esloch**
## Pull Request description

Solving issue #32.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,104 +0,0 @@
-# Byte-compiled / optimized / DLL files
```
Comment (xmnlab) [inline .gitignore:1]: any reason for removing this file?

File: cookiecutter.json [hunk 2]
```diff
@@ -15,6 +15,10 @@
     "GNU General Public License v3",
     "Other"
   ],
+  "build-system": [
```
Comment (xmnlab) [inline cookiecutter.json:18]: :+1:

Comment (xmnlab) [reply cookiecutter.json:18]: oops ... it should be `build_system`

File: {{cookiecutter.project_slug}}/pyproject.toml [hunk 3]
```diff
@@ -22,7 +22,12 @@ multi_line_output = 3
 include_trailing_comma = true
 skip_glob = ["docs/*", "*.egg-info"]
 
+{% if cookiecutter.build-system == 'poetry' -%}
```
Comment (xmnlab) [inline {{cookiecutter.project_slug}}/pyproject.toml:25]:
```suggestion
{% if cookiecutter.build_system == 'poetry' -%}
```

File: {{cookiecutter.project_slug}}/pyproject.toml [hunk 4]
```diff
@@ -22,7 +22,12 @@ multi_line_output = 3
...
 [tool.poetry]
+{% elif cookiecutter.build-system == 'pdm' -%}
```
Comment (xmnlab) [inline {{cookiecutter.project_slug}}/pyproject.toml:27]:
```suggestion
{% elif cookiecutter.build_system == 'pdm' -%}
```

File: .gitignore (hunk 1 above)
Comment (xmnlab) [reply .gitignore:1]: ping

**Comment by xmnlab** [comment]
@esloch could you create a new smoke test and call it on CI?
this smoke test would test the pdm option ... maybe you will need to update the smoke test base.

maybe with some env variable to switch between poetry and pdm ... 
let me know if you need a call for discuss that.

thanks!!

File: .gitignore (hunk 1 above)
Comment (esloch) [reply .gitignore:1]: no, es cosa de principiante...rs
Sorry!

**Comment by esloch** [comment]
> 

Hi @xmnlab, if you can, we will have a meeting to discuss this topic.

**Comment by xmnlab** [comment]
sure, feel free to send me an invitation to my google calendar. thanks!

File: tests/smoke/pdm.sh [hunk 5]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env bash
+
+SMOKE_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+. ${SMOKE_DIR}/base.sh build_system=pdm
```
Comment (xmnlab) [inline tests/smoke/pdm.sh:4]: we change the call for this base.sh .. could you please rebase your code and update this line. 

```suggestion
. ${SMOKE_DIR}/base.sh build_system pdm
```

**Comment by xmnlab** [comment]
thanks for working on that @esloch !
I will close this PR for now

**Pull request closed without merging**